tests/models/test_evaluation.py

```diff
@@ -713,21 +786,178 @@ def test_evaluator_interface(multiclass_logistic_regressor_model_uri, iris_datas
                 )
 
 
-def test_evaluate_with_multi_evaluators(multiclass_logistic_regressor_model_uri, iris_dataset):
+@pytest.mark.parametrize(
+    "baseline_model_uri,expected_error,valid_baseline_model_uri",
+    [
+        ("None", None, False),
+        ("multiclass_logistic_regressor_baseline_model_uri_4", None, True),
+        (
+            "pyfunc",
+            pytest.raises(
+                ValueError,
+                match="The baseline model argument must be a string "
+                + "URI referring to an MLflow model",
+            ),
+            False,
+        ),
+        (
+            "invalid",
+            pytest.raises(OSError, match="No such file or directory: 'invalid_uri'"),
+            False,
+        ),
+    ],
+    indirect=["baseline_model_uri"],
+)
+def test_evaluator_validation_interface(
+    multiclass_logistic_regressor_model_uri,
+    iris_dataset,
+    baseline_model_uri,
+    expected_error,
+    valid_baseline_model_uri,
+):
+    with mock.patch.object(
+        _model_evaluation_registry, "_registry", {"test_evaluator1": FakeEvauator1}
+    ):
+        evaluator1_config = {"config": True}
+        evaluator1_return_value = EvaluationResult(metrics={}, artifacts={})
+        if baseline_model_uri:
+            evaluator1_config_baseline = dict(evaluator1_config)
+            evaluator1_config_baseline.update({"is_baseline_model": True})
+        with mock.patch.object(
+            FakeEvauator1, "can_evaluate", return_value=True
+        ) as mock_can_evaluate, mock.patch.object(
+            FakeEvauator1, "evaluate", return_value=evaluator1_return_value
+        ) as mock_evaluate:
+            classifier_model = mlflow.pyfunc.load_model(multiclass_logistic_regressor_model_uri)
+            baseline_model = None
+            with mlflow.start_run() as run:
+                if valid_baseline_model_uri:
```

Comment:
Can we split this if / elif / else cases into separate tests to make each test more modular and debuggable?